clicache/src/Cache.cpp

```diff
@@ -215,12 +233,18 @@ namespace Apache
             GC::KeepAlive(m_nativeptr);
           }
 
-        _GF_MG_EXCEPTION_CATCH_ALL2
+        }
+        catch (const apache::geode::client::Exception& ex) {
+          throw Apache::Geode::Client::GeodeException::Get(ex);
+        }
+        catch (System::AccessViolationException^ ex) {
```

Comment:
Any idea why this managed exception is specifically caught only to be rethrown?

Reply:
Interesting question that deserves some thought. AccessViolations are baaaad. So you don't want to just eat them. OK, good, we don't. We rethrow them. But why catch them in the first place?

The only thing I can come up with is to help track down bugs in our code. We catch/rethrow AccessViolations all over the place (212 places), so it allows setting breakpoints which can be caught in the debugger so we can look around.

This may have been useful early on, but I doubt it provides any value now. I'm good with removing: catch (System::AccessVilationException) everywhere.